packages/livekit-rtc/src/participant.ts

```diff
@@ -349,6 +371,163 @@ export class LocalParticipant extends Participant {
+   * @param responseTimeoutMs - Timeout for receiving a response after initial connection
+   * @returns A promise that resolves with the response payload or rejects with an error.
+   * @throws Error on failure. Details in `message`.
+   */
+  async performRpc(
+    destinationIdentity: string,
+    method: string,
+    payload: string,
+    responseTimeoutMs?: number,
+  ): Promise<string> {
+    const req = {
+      localParticipantHandle: this.ffi_handle.handle,
+      destinationIdentity,
+      method,
+      payload,
+      responseTimeoutMs,
+    } as PerformRpcRequest;
+
+    const res = FfiClient.instance.request<PerformRpcResponse>({
+      message: { case: 'performRpc', value: req },
+    });
+
+    const cb = await FfiClient.instance.waitFor<PerformRpcCallback>((ev) => {
+      return ev.message.case === 'performRpc' && ev.message.value.asyncId === res.asyncId;
+    });
+
+    if (cb.error) {
+      throw RpcError.fromProto(cb.error);
+    }
+
+    return cb.payload;
+  }
+
+  /**
+   * Establishes the participant as a receiver for calls of the specified RPC method.
+   * Will overwrite any existing callback for the same method.
+   *
+   * @param method - The name of the indicated RPC method
+   * @param handler - Will be invoked when an RPC request for this method is received
+   * @returns A promise that resolves when the method is successfully registered
+   *
+   * @example
+   * ```typescript
+   * room.localParticipant?.registerRpcMethod(
+   *   'greet',
+   *   async (requestId: string, callerIdentity: string, payload: string, responseTimeoutMs: number) => {
+   *     console.log(`Received greeting from ${callerIdentity}: ${payload}`);
+   *     return `Hello, ${callerIdentity}!`;
+   *   }
+   * );
+   * ```
+   *
+   * The handler receives the following parameters:
+   * - `requestId`: A unique identifier for this RPC request
+   * - `callerIdentity`: The identity of the RemoteParticipant who initiated the RPC call
+   * - `payload`: The data sent by the caller (as a string)
+   * - `responseTimeoutMs`: The maximum time available to return a response
+   *
+   * The handler should return a Promise that resolves to a string.
+   * If unable to respond within `responseTimeoutMs`, the request will result in an error on the caller's side.
+   *
+   * You may throw errors of type `RpcError` with a string `message` in the handler,
+   * and they will be received on the caller's side with the message intact.
+   * Other errors thrown in your handler will not be transmitted as-is, and will instead arrive to the caller as `1500` ("Application Error").
+   */
+  registerRpcMethod(
+    method: string,
+    handler: (
+      requestId: string,
+      callerIdentity: string,
+      payload: string,
+      responseTimeoutMs: number,
+    ) => Promise<string>,
+  ) {
+    this.rpcHandlers.set(method, handler);
+
+    const req = {
+      localParticipantHandle: this.ffi_handle.handle,
+      method,
+    } as RegisterRpcMethodRequest;
```

Comment:
would be good to avoid this type cast and use the protobuf creation scheme instead. 

```
req: RegisterRpcMethodRequest = create(RegisterRpcMethodRequestSchema, { ... });

Reply:
yep - thank you